[master] Handle regular expressions which do not not use grouping

vin01 opened the pull request:
### What does this PR do?

Handle regular expressions which do not not use grouping in `regex_search` and `regex_match`


### Previous Behavior

```
FAILED tests/pytests/functional/modules/state/test_jinja_filters.py::test_filter[regex_match_ungrouped0] - assert {'ret': '()'} == {'ret': "('1.abc',)"}
FAILED tests/pytests/functional/modules/state/test_jinja_filters.py::test_filter[regex_match_ungrouped1] - assert {'ret': '()'} == {'ret': "('1.abc',)"}
```

### New Behavior

Entire match is returned as a tuple.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

dwoz reviewed (changes requested):
Looks okay to me but we'll need a test case and changelog. Ideally there would also be an open issue with an explanation of what we're fixing.

vin01 committed:
Add tests for ungrouped regex match and search

vin01 commented:
> Looks okay to me but we'll need a test case and changelog. Ideally there would also be an open issue with an explanation of what we're fixing.

I had added [a couple](https://github.com/saltstack/salt/pull/65708/commits/fd4f6d2a704f733185f1661ccd848c8405667c5f) of functional state tests already, wouldn't that fly? :)

Added an issue and corresponding changelog entry, I skipped it initially as I presumed this behavior might be intentional and up for a bit of discussion.

vin01 committed:
Use .group() to return ungrouped regex matches

Currently if a regex does not use grouping using parentheses, no match is returned despite a valid regex match taking place. This is not accurate since using grouping should not be necessary.

dwoz reviewed (approved):
@vin01 

Approved, but also asking for @whytewolf and @s0undt3ch to have a look.

vin01 commented:
this branch is getting out of date, please feel free to update and merge if you think this is ready now.

vin01 commented:
can this be merged please?
